reworked Label class

leezh opened the pull request:
- no longer inherits Range - instead, more sensible
  function names controlling lines visible
- more accurate vertical alignment
- percent_visible preserved even after setting new text

Largely motivated by the frustratingly inaccurate vertical alignment, particularly with large fonts.
Before: https://gfycat.com/WaterySimpleHornedtoad
After: https://gfycat.com/NeighboringMetallicKilldeer

The code for controlling the lines visible was already there, but I changed it so that instead of using `set_val()` and `set_max()` you now use `set_lines_skipped()` and `set_max_lines_visible()`.

Also, `percent_visible` and `visible_chars` are now recalculated properly each time you use `set_text()` because that could change the total length, and when you use `set_visible_characters()`.

leezh committed:
reworked Label class

 - no longer inherits Range - instead, more sensible
   function names controlling lines visible
 - more accurate vertical alignment
 - percent_visible preserved even after setting new text

okamstudio commented:
very appreciated! will hopefully merge soon

On Mon, Sep 7, 2015 at 4:39 PM, Lee Zher Huei <notifications@github.com>
wrote:

>
>    - no longer inherits Range - instead, more sensible function names
>    controlling lines visible
>    - more accurate vertical alignment
>    - percent_visible preserved even after setting new text
>
> Largely motivated by the frustratingly inaccurate vertical alignment,
> particularly with large fonts.
> Before: https://gfycat.com/WaterySimpleHornedtoad
> After: https://gfycat.com/NeighboringMetallicKilldeer
>
> The code for controlling the lines visible was already there, but I
> changed it so that instead of using set_val() and set_max() you now use
> set_lines_skipped() and set_max_lines_visible().
>
> Also, percent_visible and visible_chars are now recalculated properly
> each time you use set_text() because that could change the total length,
> and when you use set_visible_characters().
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/okamstudio/godot/pull/2449
> Commit Summary
>
>    - reworked Label class
>
> File Changes
>
>    - *M* scene/gui/label.cpp
>    <https://github.com/okamstudio/godot/pull/2449/files#diff-0> (250)
>    - *M* scene/gui/label.h
>    <https://github.com/okamstudio/godot/pull/2449/files#diff-1> (39)
>
> Patch Links:
>
>    - https://github.com/okamstudio/godot/pull/2449.patch
>    - https://github.com/okamstudio/godot/pull/2449.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2449>.
>



-- 

*OkamStudio*

leezh committed:
added get_visible_characters() to Label

Ovnuniarchos commented:
Nice! And a remainder to update docs when this gets merged…

leezh committed:
fix minsize-related issues with Label

 - fixed end of string adding to Label minsize
 - exposed set_clip_text() and is_clipping_text() to the script side
 - text would now never stick outside its boundaries
 - label min-height is now restricted by max_lines_visible

leezh commented:
Did more experimenting and pushed out a couple of changes.

leezh committed:
updated Label documentation

bojidar-bg commented:
:clock1030:

reduz commented:
@akien-mga should I merge or wait for the doc to be updated somewhere else?

akien-mga commented:
@reduz I think you can merge, the only pending PR for documentation update is #2514, but I don't think it would conflict.

jackmakesthings commented:
:+1:

reduz commented:
test this well, guys :)

On Fri, Sep 25, 2015 at 11:28 AM, jack <notifications@github.com> wrote:

> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/okamstudio/godot/pull/2449#issuecomment-143238223>.
>
